plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -361,8 +363,16 @@ public void removeInstanceSnapshot(InstanceKey instanceKey) throws SnapshotExcep
         // use registry API directly because docker doesn't have such API yet
         // https://github.com/docker/docker-registry/issues/45
         final DockerInstanceKey dockerInstanceKey = new DockerInstanceKey(instanceKey);
-        String registry = dockerInstanceKey.getRegistry();
-        String repository = dockerInstanceKey.getRepository();
+        if (!snapshotUseRegistry) {
```

Comment:
Value of this property can be changed between calls of snaspshot saving and removal.